src/vs/editor/contrib/codeAction/browser/codeActionUi.ts

```diff
@@ -23,6 +23,7 @@ export class CodeActionUi extends Disposable {
 	private readonly _codeActionWidget: Lazy<CodeActionMenu>;
 	private readonly _lightBulbWidget: Lazy<LightBulbWidget>;
 	private readonly _activeCodeActions = this._register(new MutableDisposable<CodeActionSet>());
+	private previewOn: boolean = false;
```

Comment:
Is there a way to do this without introducing state?

Reply:
Since it defaults to having preview off, I thought i'd just default it false (set true when the specific keybinding is pressed)

Reply:
State adds complexity so best to avoid it if possible. Can you pass in `previewOn` into that function instead of setting state that the function uses?